### PR TITLE
GH-12630: Improve address book events

### LINKED ISSUE #12630: Improve address book events
see if we can use event builder pattern

See first iteration PR is #12427

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: projects/core/src/checkout/events/checkout-event.listener.ts [hunk 1]
```diff
@@ -3,10 +3,9 @@ import { Subscription } from 'rxjs';
 import { filter } from 'rxjs/operators';
 import { EventService } from '../../event/event.service';
 import {
-  UserAddressDeleteEvent,
```
Comment (plabadie) [inline projects/core/src/checkout/events/checkout-event.listener.ts]: This deprecated file will be deleted on epic branch.

File: projects/core/src/checkout/events/checkout-event.listener.spec.ts [hunk 2]
```diff
@@ -3,9 +3,8 @@ import { BehaviorSubject, Observable } from 'rxjs';
 import { CxEvent } from '../../event/cx-event';
 import { EventService } from '../../event/event.service';
 import {
-  UserAddressDeleteEvent,
```
Comment (plabadie) [inline projects/core/src/checkout/events/checkout-event.listener.spec.ts]: This deprecated file will be deleted on epic branch.

File: projects/core/src/user/events/user.events.ts [hunk 3]
```diff
@@ -3,22 +3,17 @@ import { Address } from '../../model/address.model';
 
 export abstract class UserAddressEvent extends CxEvent {}
 
-export class UserAddressUpdateEvent extends UserAddressEvent {
-  static readonly type = 'UserAddressUpdateEvent';
+export class UpdateUserAddressEvent extends UserAddressEvent {
+  static readonly type = 'UpdateUserAddressEvent';
   addressId: string;
```
Comment (WeizhengSap) [inline projects/core/src/user/events/user.events.ts:9]: can we just pass `addressId`?

Comment (plabadie) [reply projects/core/src/user/events/user.events.ts:9]: I don’t think we have control over the payload.  From the tests I did so far, the action payload is copied to the event when using the event builder pattern

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**